### DISCUSSION TITLE
What is the recommended way to rollback to the last deploy?

### DISCUSSION THREAD

**Discussion by followbl** [vercel/vercel#5209] [Help] [answered]
I see the rollback functionality built in to the vercel/github plugin (which calls for a revert commit - and then it'll rollback to the last deploy) - is there an equivalent within the CLI or a couple of other commands?

**Comment by paulogdm** [comment]
EDIT: see https://github.com/vercel/vercel/discussions/5209#discussioncomment-6520953

---

_Old answer_:
You can either issue a git action or use the dashboard with "promote to production". We are working on a better DX for this specific use-case as well.

**Reply by followbl** [reply]
thanks for the reply - I'm trying to find via the docs or the cli help - is there a way to get the currently assigned/aliases deploy via the cli - we used to be able to do this.

**Comment by zzoman** [comment]
Is there any update of this?
There is some problems which are making rollback difficult.

- Can't list up only production deployment on vercel dashboard.
- Rollback is an important part of deployment but I can't find any documents about it.

**Reply by leerob** [reply]
Yes, we are actively working on this, including better documentation!

**Reply by SindreSvendby** [reply]
> Instant rollbacks when reverting changes assigned to a custom domain 
(Source: https://vercel.com/docs/concepts/git)

We tested this by doing a git revert and pushing it to the production branch. The deploy was not instant. it created a new build/deployment. 

Are you able to provide better insight in how this should behave?

**Reply by leerob** [reply]
You can currently do an instant rollback with `vercel alias` - https://vercel.com/docs/cli#commands/alias

We are working to add this option directly to the UI as well.

**Reply by aseemk** [reply]
@leerob: Could you share more about how to achieve the instant rollback with `vercel alias`? I've run `vercel alias ls` but am not seeing any of my project's actual custom domains in the output — only `.vercel.app` domains (under both `source` and `url`).

Thank you!

**Reply by leerob** [reply]
`vercel alias set [deployment-url] [custom-domain]` should do it!

**Reply by Chorany** [reply]
Hi, is there any update as to the status of adding this to the UI?

**Reply by molebox** [reply]
Hey @Chorany this is being worked on as we speak! Thanks for your patience 🙏🏻

**Reply by joshcawthorne** [reply]
Any ETA on this @molebox? 🙏

**Reply by jzxhuang** [reply]
@leerob @molebox @paulogdm 

Also interested in any updates/eta here! I've tried out the suggested `vercel alias set [deployment-url] [custom-domain]` but acutally running into an issue on the latest CLI (28.4.10) when using custom domains:

- `vercel alias set [deployment-url] *.vercel.app` works
- `vercel alias set [deployment-url] [custom-domain]` doesn't.

running this command with `--debug`, I can see this API request failing: `https://api.vercel.com/v4/domains/<custom-domain>?teamId=<teamId>`. It always returns 403 Forbidden, with the CLI error message: `Error: You don't have access to the domain <customDomain> under <team>.`. This is really confusing: I don't even see this endpoint documented in the Vercel API docs at all! And, running `vercel alias ls` clearly shows that I do indeed own that domain 🤔 

And additionally, if I manually call the endpoint toset an alias with the custom domain, it _does_ work! POST `https://api.vercel.com/v2/deployments/<deployment_id>/aliases?teamId=<teamId>` with body `{ alias: <customDomain> }`

Fundamentally, I think the bigger root problem is actually the fact that we are unable to promote a production deployment for builds from the CLI (see https://vercel.com/docs/errors#error-list/non-git-deployment-cannot-be-promoted-to-production-from-dashboard). This `vercel alias set` is a hacky workaround, and it's quite confusing to me why we can't promote a CLI build to production? It's quite an unsafe workflow to be building _and_ deploying to produciton at the same time. What about when you want to rollback? You gotta go re-run the build step via CLI for that specific commit, that's gonna take 20+ minutes!!

Thanks for any help you can provide!

**Comment by philwolstenholme** [comment, marked as answer]
https://vercel.com/docs/cli/rollback – for anyone who finds this in 2023 and beyond (also available via the UI: https://vercel.com/docs/concepts/deployments/instant-rollback )
